src/coap_mbedtls.c

```diff
@@ -59,6 +60,13 @@
 #define mbedtls_strdup(a) strdup(a)
 #define mbedtls_strndup(a,b) strndup(a,b)
 
+#ifndef MBEDTLS_KEY_EXCHANGE__SOME__PSK_ENABLED
```

Comment:
The double underscore seems to be intentional. However, this is a bit confusing; is there a particular reason to have these two almost identical symbols?

Reply:
That's more a question for the mbedtls project. AFAIK, currently `MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED` is sued, but before the other one was used (even, if I don't know on which version :-).

Therefore, coap_mbedtls.c, line 63:

```
#ifndef MBEDTLS_KEY_EXCHANGE__SOME__PSK_ENABLED
/* definition changed in later mbedtls code versions */
#ifdef MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED
#define MBEDTLS_KEY_EXCHANGE__SOME__PSK_ENABLED
#endif /* MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED */
#endif /* ! MBEDTLS_KEY_EXCHANGE__SOME__PSK_ENABLED */

```

Reply:
Okay, thanks for the clarification.

Reply:
See previous response - they went from 2 _ to 1 _ somewhere between 2.16.5 and 2.25.0.

Code unchanged as MBEDTLS_KEY_EXCHANGE__SOME__PSK_ENABLED is the main code wrapper checker for PSK functionality (needed for esp-idf project).